src/inet/TCPEndPoint.cpp

```diff
@@ -277,10 +277,14 @@ INET_ERROR TCPEndPoint::Listen(uint16_t backlog)
         res = chip::System::MapErrorPOSIX(errno);
 
     // Wait for ability to read on this endpoint.
-    mRequestIO.SetRead();
+    mWatcher.Open(mSocket);
+    mWatcher.SetCallback(HandlePendingIO, this);
+    mWatcher.SetReadable();
 
+#if CHIP_SYSTEM_CONFIG_USE_IO_THREAD
```

Comment:
It's unclear from the current arrangement of the code whether CHIP_SYSTEM_CONFIG_USE_IO_THREAD == (CHIP_SYSTEM_CONFIG_USE_SOCKETS == 1) && (CHIP_SYSTEM_CONFIG_SOCKETS_USE_SELECT == 0) or if CHIP_SYSTEM_CONFIG_USE_IO_THREAD is completely independent of either of those.

Reply:
Yes, this is a part that needs to be fleshed out. It's independent of the interface for file descriptor watching, but dependent on the implementation. Some may require a thread (as select does), some may require there not be one (e.g. using a CHIP-consuming application's mechanism instead), and some may go either way. Actual `#if` tests should end up much reduced, since I can probably abstract `WakeIOThread()` (previously `WakeSelect()`) a bit more.